Dockerfile.full_ros_build

```diff
@@ -1,46 +1,44 @@
-# This dockerfile expects proxies to be set via --build-arg if needed
-# It also expects to be contained in the /navigation2 root folder for file copy
+# This dockerfile expects to be contained in the /navigation2 root folder for file copy
 #
 # Example build command:
-# export CMAKE_BUILD_TYPE=Debug
-#
 # This determines which version of the ROS2 code base to pull
 # export ROS2_BRANCH=master
-#
-# This determines which ros2 docker image to use as a base for getting ros2_dependencies
-# The latest (osrf/ros2:nightly) is usually the best choice unless a dependency
-# has been removed that an older codebase depends on. In those cases, using
-# ros:DISTRO-ros-core would be a good choice where distro is crystal, dashing, etc.
-# export ROS2_DOCKER_IMAGE=osrf/ros2:nightly
 # docker build -t nav2:full_ros_build
-#   --build-arg CMAKE_BUILD_TYPE \
 #   -f Dockerfile.full_ros_build ./
-
-# We're only building on top of a ROS docker image to get the basics
+#
+# We're only building on top of a ros2 devel image to get the basics
 # prerequisites installed such as the apt source, rosdep, etc. We don't want to
 # actually use any of the ros dashing packages. Instead we are going to build
 # everything from source in one big workspace.
-ARG FROM_IMAGE=osrf/ros2:nightly
+
+ARG FROM_IMAGE=osrf/ros2:devel
 FROM $FROM_IMAGE
 
-# install ROS2 dependencies
+# install packages
 RUN apt-get update && apt-get install -q -y \
-      wget \
-      libasio-dev \
+    libasio-dev \
+    libtinyxml2-dev \
+    wget \
     && rm -rf /var/lib/apt/lists/*
 
-# setup directory structure
-ENV ROS2_WS /opt/ros2_ws
-RUN mkdir -p $ROS2_WS/src
+ARG ROS2_BRANCH=master
+ENV ROS2_BRANCH=$ROS2_BRANCH
+ENV ROS_VERSION=2 \
+    ROS_PYTHON_VERSION=3
+
 WORKDIR $ROS2_WS
 
-# clone ROS 2 packages
+RUN wget https://raw.githubusercontent.com/ros2/ros2/$ROS2_BRANCH/ros2.repos \
+    && vcs import src < ros2.repos
 
-ARG ROS2_BRANCH=master
-RUN wget https://raw.githubusercontent.com/ros2/ros2/$ROS2_BRANCH/ros2.repos
-RUN vcs import src < ros2.repos
+# install ros2 dependencies
```

Comment:
We don't need this rosdep here, since all the ros 2 dependencies will get picked up when we re-run rosdep below. Are you doing it here to help improve layer caching?

Reply:
> Are you doing it here to help improve layer caching?

Yea, so as long as you don't touch the .repo file, you won't have to install/rebuild 
 basic dependencies.